### PR TITLE
make_date_list working now

### PR DESCRIPTION
**Pull request by yossiyossi**
The make_date_list function seems to be working now, for initial example.
Still To Do: More Tests, Creation of _lectures/ directory and creation of lecture file stubs with correct dates
EDIT: this is now outdated, see new comment (below)

### CODE REVIEW

File: lecture-gen.py [hunk 1]
```diff
@@ -27,38 +31,112 @@ def validate_start_date(date):
...
+    """
+    Return a tuple of two datetimes, one for each lecture day
+    start_datetime is the first lecture day of the week, the second
+    is calculated by start_datetime + 2 days days_ahead
+
+    Warning: Does not account for holidays (see days_without_holidays)
+    """
+
+    day_one = start_datetime
+    day_two = n_days_ahead(start_datetime, 2)
```
Comment (ykharitonova) [inline lecture-gen.py]: Seems like this assumes that lectures are 2 days apart. However, a course could be on TWR or any other combination of days. Perhaps, we can have `lecture_days` that is a subset of `legal_days_of_week`? Then, we'd parse that variable to know what offset to use.

File: lecture-gen.py [hunk 2]
```diff
@@ -27,38 +31,112 @@ def validate_start_date(date):
...
+    return (day_one, day_two)
+
+def days_without_holidays(days_of_week, holiday_list):
+    """
+    Given a tuple for the two lecture days of the week, returns a list
```
Comment (ykharitonova) [inline lecture-gen.py:70]: It would be nice to have flexibility about the number of lecture days per week.

File: lecture-gen.py [hunk 3]
```diff
@@ -27,38 +31,112 @@ def validate_start_date(date):
...
+    is calculated by start_datetime + 2 days days_ahead
+
+    Warning: Does not account for holidays (see days_without_holidays)
+    """
+
+    day_one = start_datetime
+    day_two = n_days_ahead(start_datetime, 2)
+    return (day_one, day_two)
+
+def days_without_holidays(days_of_week, holiday_list):
```
Comment (ykharitonova) [inline lecture-gen.py:68]: Can the holiday list be parsed from the following format (that's currently used in _config.yml)?
```json
  [
  {"label":"First day of classes", "date":"2019-01-07" },
  {"label":"Univ Holiday", "date":"2019-01-21" },
  {"label":"Drop deadline (no W grade)","date":"2019-02-04" },
  {"label":"Univ Holiday", "date":"2019-02-18" },
  {"label":"W19 Instruction Ends", "date":"2019-03-15" },  
  {"label":"W19 Quarter Ends", "date":"2019-03-22" }          
  ]
```

**Review by ykharitonova — CHANGES_REQUESTED** [review]
A few considerations to keep in mind as we try to make the script flexible and generalizable.

File: lecture-gen.py (hunk 1 above)
Comment (yossiyossi) [reply lecture-gen.py]: I added support for any number of days in the newest commit in this pull request.

File: lecture-gen.py (hunk 2 above)
Comment (yossiyossi) [reply lecture-gen.py:70]: @ykharitonova Just added support for any days of the week in the newest pull request.

**Comment by yossiyossi** [comment]
Hi Professors,
Please see the updated lecture_gen function in the newest commit which I believe is working.
Still Todo: try to parse holiday list from the format shown above in @ykharitonova 's comment.
I was not sure about how to handle the path of the directory to create, so I currently just take it in as a parameter to lecture_gen. The code now uses the current working directory and creates the _lectures directory in it to test.
Please let me know of any changes/additions you would like to see!

**Pull request merged**